### ISSUE TITLE
the length of usernames and passwords must match with 1.8.2

### ISSUE THREAD

**Issue by jbgomond** [aquasecurity/harbor-scanner-trivy#359]
**What steps did you take and what happened:**

I updated docker images from 1.8.0 to 1.8.2 (all docker harbor containers).
I'm now having this error when I launch an analysis from harbor, using trivy.
The service health is OK.

> 2023-07-08T19:48:55+02:00 [INFO] [/pkg/scan/job.go:387]: {
  "uuid": "02c8fdac-abdc-11ed-bf71-0242ac130008",
  "name": "Local Trivy",
  "description": "",
  "url": "http://localhost:8081",
  "disabled": false,
  "is_default": true,
  "health": "healthy",
  "auth": "",
  "access_credential": "[HIDDEN]",
  "skip_certVerify": false,
  "use_internal_addr": false,
  "adapter": "Trivy",
  "vendor": "Aqua Security",
  "version": "Unknown",
  "create_time": "2023-02-13T20:21:34.62387Z",
  "update_time": "2023-07-08T17:35:01.05781Z"
}
2023-07-08T19:48:55+02:00 [INFO] [/pkg/scan/job.go:387]: {
  "registry": {
    "url": "https://XXXX",
    "authorization": "[HIDDEN]"
  },
  "artifact": {
    "namespace_id": 2,
    "repository": "XXX/XXX",
    "tag": "8.1",
    "digest": "sha256:9b80c586e892182fe126c19382737c39334942b2a1ac412d56b64dc4277da16b",
    "mime_type": "application/vnd.docker.distribution.manifest.v2+json"
  }
}
2023-07-08T19:48:55+02:00 [INFO] [/pkg/scan/job.go:167]: Report mime types: [application/vnd.security.vulnerability.report; version=1.1]
2023-07-08T19:48:55+02:00 [INFO] [/pkg/scan/job.go:224]: Get report for mime type: application/vnd.security.vulnerability.report; version=1.1
2023-07-08T19:48:57+02:00 [ERROR] [/pkg/scan/job.go:294]: check scan report with mime type application/vnd.security.vulnerability.report; version=1.1: running trivy wrapper: running trivy: exit status 1: 2023-07-08T19:48:57.341+0200	[31mFATAL[0m	flag error:
    github.com/aquasecurity/trivy/pkg/commands.NewImageCommand.func2
        /bitnami/blacksmith-sandox/trivy-0.42.0/src/github.com/aquasecurity/trivy/pkg/commands/app.go:312
   registry flag error:
    github.com/aquasecurity/trivy/pkg/flag.(*Flags).ToOptions
        /bitnami/blacksmith-sandox/trivy-0.42.0/src/github.com/aquasecurity/trivy/pkg/flag/options.go:484
   the length of usernames and passwords must match:
    github.com/aquasecurity/trivy/pkg/flag.(*RegistryFlagGroup).ToOptions
        /bitnami/blacksmith-sandox/trivy-0.42.0/src/github.com/aquasecurity/trivy/pkg/flag/registry_flags.go:68
: general response handler: unexpected status code: 500, expected: 200

It seems to be linked with that change:  https://github.com/aquasecurity/trivy/pull/3906.

**What did you expect to happen:**

Continue working as it was with the new version

**Anything else you would like to add:**

**Environment:**

- Harbor version: 1.8.2
- Harbor Scanner Adapter for Trivy version: v0.30.12 | v0.42.0
- Harbor installation process ([Installer script][harbor-installer], [Helm chart][harbor-helm], etc.): Docker

[harbor-helm]: https://goharbor.io/docs/2.3.0/install-config/harbor-ha-helm/
[harbor-installer]: https://goharbor.io/docs/2.3.0/install-config/download-installer/

**Comment by piyush94** [comment]
We're also facing the same issue.

- Harbor version: 2.6.3
- Harbor Scanner Adapter for Trivy version: v0.30.12 | v0.42.0
- Harbor installation process ([Installer script](https://goharbor.io/docs/2.3.0/install-config/download-installer/), [Helm chart](https://goharbor.io/docs/2.3.0/install-config/harbor-ha-helm/), etc.): Docker

**Comment by horeyes** [comment]
Also having this issue:

Harbor version: 2.7
Harbor Scanner Adapter: v0.30.15 
Trivy: v0.44
Installation process: Helm Chart

**Comment by piyush94** [comment]
@ByJacob I'll have to disagree here. I have two instances of Harbor and both have spaces in the Interrogation Service name.
The scanner is working in one instance and not working in the other one.

Instance one (not working):
![image](https://github.com/aquasecurity/harbor-scanner-trivy/assets/11614204/578e0e2b-eaec-442a-b5b2-f607dbab65da)

Instance two (working):
![image](https://github.com/aquasecurity/harbor-scanner-trivy/assets/11614204/d23c0437-b5dc-46f2-bae2-a1559457801a)

**Comment by piyush94** [comment]
@ByJacob I tried removing the space from the Interrogation scanner name and it did help. The scanner is working now. So, thanks for the suggestion. I'll definitely try out the debug script to see what is the value of the variables in my second instance.

**Comment by piyush94** [comment]
@ByJacob Got the error in second instance as well. It was re-installed a few days ago so maybe that cleared the scan cache. Earlier I was not seeing the error here. So, I removed the space here as well and it's fine now.
Again thanks for finding the cause and suggesting the work around.

**Comment by jbgomond** [comment]
Indeed, removing the space in the scanner name is fixing the issue. It seems there's a bug in that area.
